tests/src/Kernel/SchemaFactoryTest.php

```diff
@@ -0,0 +1,175 @@
+<?php
+
+namespace Drupal\Tests\schemata\Kernel;
+
+use Drupal\KernelTests\KernelTestBase;
+use Drupal\node\Entity\NodeType;
+use Drupal\schemata\SchemaFactory;
+use Drupal\schemata\Schema\SchemaInterface;
+use Drupal\schemata\Schema\NodeSchema;
+
+/**
+ * Tests the Schema Factory service.
+ *
+ * @coversDefaultClass \Drupal\schemata\SchemaFactory
+ * @group Schemata
+ * @group SchemataCore
+ */
+class SchemaFactoryTest extends KernelTestBase {
+
+  /**
+   * Schema Factory.
+   *
+   * @var \Drupal\schemata\SchemaFactory
+   */
+  protected $factory;
+
+  /**
+   * {@inheritdoc}
+   */
+  public static $modules = [
+    'schemata',
+    'field',
+    'node',
+    'serialization',
+    'system',
+    'user',
+  ];
+
+  /**
+   * {@inheritdoc}
+   */
+  protected function setUp() {
+    parent::setUp();
+
+    // Add the entity schemas.
+    $this->installEntitySchema('node');
+    $this->installEntitySchema('user');
+    // Add the additional table schemas.
+    $this->installSchema('system', ['sequences']);
+    $this->installSchema('node', ['node_access']);
+    $this->installSchema('user', ['users_data']);
+
+    $this->nodeType = NodeType::create([
+      'type' => 'article',
+    ]);
+    $this->nodeType->save();
+
+    $this->factory = \Drupal::service('schemata.schema_factory');
+  }
+
+  /**
+   * @covers ::create
+   */
+  public function testCreateNodeBaseSchema() {
+    $schema = $this->factory->create('node');
+    $this->assertInstanceOf(SchemaInterface::class, $schema);
+    $this->assertNotInstanceOf(NodeSchema::class, $schema);
+    $this->assertSchemaHasNoBundle($schema);
+    $this->assertSchemaHasTitle($schema);
+    $this->assertSchemaHasDescription($schema);
+    $this->assertSchemaHasProperties($schema);
+  }
+
+  /**
+   * @covers ::create
+   */
+  public function testCreateNodeArticleSchema() {
+    $schema = $this->factory->create('node', 'article');
+    $this->assertInstanceOf(SchemaInterface::class, $schema);
+    $this->assertInstanceOf(NodeSchema::class, $schema);
+    $this->assertSchemaHasBundle($schema, 'article');
+    $this->assertSchemaHasTitle($schema);
+    $this->assertSchemaHasDescription($schema);
+    $this->assertSchemaHasProperties($schema);
+  }
+
+  /**
+   * @covers ::create
+   */
+  public function testCreateUserSchema() {
+    $schema = $this->factory->create('user');
+    $this->assertInstanceOf(SchemaInterface::class, $schema);
+    $this->assertNotInstanceOf(NodeSchema::class, $schema);
+    $this->assertSchemaHasNoBundle($schema);
+    $this->assertSchemaHasTitle($schema);
+    $this->assertSchemaHasDescription($schema);
+    $this->assertSchemaHasProperties($schema);
+  }
```

Comment:
Since these 3 methods are nearly identical, you might be able to utilize a [data provider](https://phpunit.de/manual/current/en/writing-tests-for-phpunit.html#writing-tests-for-phpunit.data-providers) here.

Reply:
The 3 methods above that is, not those 3 asserts.

Reply:
That's a good thought, but I expect these might be deepened in different directions, so the duplication seems like it may be a good starting point for the future.